source/simulator/solver.cc

```diff
@@ -384,19 +384,39 @@ namespace aspect
                               :
                               introspection.name_for_compositional_index(advection_field.compositional_variable) + " composition");
 
+    if (advection_field.is_temperature())
+      advection_solver_tolerance = parameters.temperature_solver_tolerance;
+    else
+      advection_solver_tolerance = parameters.composition_solver_tolerance;
+
+    const double tolerance = std::max(1e-50,
+                                      advection_solver_tolerance*system_rhs.block(block_idx).l2_norm());
+
+#if DEAL_II_VERSION_GTE(9,0,0)
+    SolverControl solver_control (1000, tolerance);
+#else
+    ExtendedSolverControl solver_control (1000, tolerance);
```

Comment:
You no longer need the distinction here (i.e., just use `SolverControl` unconditionally) because for old versions of deal.II, writing `SolverControl` finds the declaration in namespace `aspect`, whereas for new versions of deal.II, it just takes the one in namespace `dealii` because there is none in the "closer namespace" `aspect`.